src/NewsService.Validation/EditNewsRequestValidator.cs

```diff
@@ -0,0 +1,160 @@
+            context.AddFailure(propertyName, validateDelegate.Value);
+          }
+        }
+      }
+
+      #endregion
+
+      #region Paths
+
+      AddСorrectPaths(
+        new()
+        {
+          nameof(EditNewsRequest.Preview),
+          nameof(EditNewsRequest.Content),
+          nameof(EditNewsRequest.Subject),
+          nameof(EditNewsRequest.Pseudonym),
+          nameof(EditNewsRequest.AuthorId),
+          nameof(EditNewsRequest.DepartmentId),
+          nameof(EditNewsRequest.IsActive),
+        });
+
+      AddСorrectOperations(nameof(EditNewsRequest.Preview), new() { OperationType.Replace });
+      AddСorrectOperations(nameof(EditNewsRequest.Content), new() { OperationType.Replace });
+      AddСorrectOperations(nameof(EditNewsRequest.Subject), new() { OperationType.Replace });
+      AddСorrectOperations(nameof(EditNewsRequest.Pseudonym), new() { OperationType.Replace });
+      AddСorrectOperations(nameof(EditNewsRequest.AuthorId), new() { OperationType.Replace });
+      AddСorrectOperations(nameof(EditNewsRequest.DepartmentId), new() { OperationType.Replace });
+      AddСorrectOperations(nameof(EditNewsRequest.IsActive), new() { OperationType.Replace });
+
+      #endregion
+
+      #region string property
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Preview),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Preview cannot be empty." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Content),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Content cannot be empty." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Subject),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Subject cannot be empty." },
+          { x => x.value.ToString().Length < 120, "Subject is too long." },
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.Pseudonym),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrEmpty(x.value.ToString()), "Pseudonym cannot be empty." },
+          { x => x.value.ToString().Length < 30, "Pseudonym is too long." },
+        });
+
+      #endregion
+
+      #region AuthorId, DepartmentId
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.DepartmentId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => Guid.TryParse(x.value.ToString(), out Guid result), "Department id has incorrect format" }
+        });
+
+      AddFailureForPropertyIf(
+        nameof(EditNewsRequest.AuthorId),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => Guid.TryParse(x.value.ToString(), out Guid result), "Author id has incorrect format" }
```

Comment:
add request to check user existence consumer of user service

Reply:
done